posthog/cdp/templates/snapchat_ads/template_snapchat_ads.py

```diff
@@ -0,0 +1,141 @@
+from posthog.cdp.templates.hog_function_template import HogFunctionTemplate
+
+template: HogFunctionTemplate = HogFunctionTemplate(
+    status="alpha",
+    type="destination",
+    id="template-snapchat-ads",
+    name="Snapchat Ads Conversions",
+    description="Send conversion events to Snapchat Ads",
+    icon_url="/static/services/snapchat-ads.png",
+    category=["Advertisement"],
+    hog="""
+let body := {
+    'data': [
+        {
+            'event_name': inputs.eventName,
+            'action_source': inputs.actionSource,
+            'event_time': inputs.eventTime,
+            'user_data': {},
+            'custom_data': {}
+        }
+    ]
+}
+
+if (not empty(event.properties.$current_url)) body.data.1.event_source_url := event.properties.$current_url
+
+for (let key, value in inputs.userData) {
+    // e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 is an empty string hashed
+    if (not empty(value) and value != 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855') {
+        body.data.1.user_data[key] := value
+    }
+}
+
+for (let key, value in inputs.customData) {
+    // e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 is an empty string hashed
+    if (not empty(value) and value != 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855') {
+        body.data.1.custom_data[key] := value
+    }
+}
+
+let res := fetch(f'https://tr.snapchat.com/v3/{inputs.pixelId}/events?access_token={inputs.accessToken}', {
+    'method': 'POST',
+    'headers': {
+        'Content-Type': 'application/json',
+    },
+    'body': body
+})
+if (res.status >= 400) {
+    throw Error(f'Error from tr.snapchat.com (status {res.status}): {res.body}')
+}
+""".strip(),
+    inputs_schema=[
+        {
+            "key": "accessToken",
+            "type": "string",
+            "label": "Access token",
+            "description": "Check out this page on how to obtain such a token: https://developers.snap.com/api/marketing-api/Conversions-API/GetStarted#access-token",
```

Comment:
Could we not use the oauth integration? Feels way nicer https://developers.snap.com/snap-kit/login-kit/Tutorials/web

Reply:
It would be nicer, but it would require two separate oauth apps since each only supports a single redirect URL. Doesn't seem to be worth the effort as the token is pretty easy to generate, but can look into it if you think it's worth it.